src/components/EventDescription.jsx

```diff
@@ -0,0 +1,54 @@
+import { doc, getDoc } from 'firebase/firestore';
+import { useEffect, useState } from 'react';
+import { useParams, useLocation } from 'react-router';
+import { db } from '../firebaseConfig';
+
+function EventDescription() {
+  const { eventId } = useParams();
+  const location = useLocation();
+  const [event, setEvent] = useState(location.state?.event || null);
```

Comment:
why not use the useLoaderData hook from react router to fetch the data for each event when going to single event page?